python_modules/dagster/dagster/core/execution.py

```diff
@@ -186,9 +186,23 @@ def output_expectations(self):
     @property
     def success(self):
         '''Whether the solid execution was successful'''
+        any_success = False
+        for step_event in itertools.chain(
+            self.input_expectations, self.output_expectations, self.transforms
+        ):
+            if step_event.event_type == DagsterEventType.STEP_FAILURE:
+                return False
+            if step_event.event_type == DagsterEventType.STEP_SUCCESS:
+                any_success = True
+
+        return any_success
+
```

Comment:
are these the right semantics for `success` and `skipped` on `SolidExecutionResult`?

`success` if no failures and at least a success (ie not all skipped)
`skipped` if all skipped

Reply:
I think that sounds about right. Let's see if it ends up with any counterintuitive behavior.